project_restorer/src/main/scala/PlutoCoreMessageProcessor.scala

```diff
@@ -57,34 +59,38 @@ class PlutoCoreMessageProcessor(mxsConfig:MatrixStoreConfig)(implicit mat:Materi
-      (allResults.head, allResults(1)) match {
-        case (Right(nearlineResults), Right(onlineResults)) =>
-          if (nearlineResults.length < 10000 && onlineResults.length < 10000) {
-            logger.info(s"About to send bulk messages for ${nearlineResults.length} nearline results")
-            framework.bulkSendMessages(routingKey, nearlineResults)
-            logger.info(s"About to send bulk messages for ${onlineResults.length} online results")
-            framework.bulkSendMessages(routingKey, onlineResults)
-            logger.info(s"Bulk messages sent; about to send the RestorerSummaryMessage for project ${updateMessage.id}")
-            val msg = RestorerSummaryMessage(updateMessage.id, ZonedDateTime.now(), updateMessage.status, numberOfAssociatedFilesNearline = nearlineResults.length, numberOfAssociatedFilesOnline = onlineResults.length)
-            Right(MessageProcessorConverters.contentToMPRV(msg.asJson))
-          } else {
-            throw new RuntimeException(s"Too many files attached to project ${updateMessage.id}, nearlineResults = ${nearlineResults.length}, onlineResults = ${onlineResults.length}")
+
+  def handleUpdateMessage(updateMessage: ProjectUpdateMessage, routingKey: String, framework: MessageProcessingFramework): Future[Either[String, MessageProcessorReturnValue]] =
+    updateMessage.status match {
+      case status if statusesMediaNotRequired.contains(status.toLowerCase)  =>
+        Future.sequence(Seq(getNearlineResults(updateMessage.id), getOnlineResults(updateMessage.id))).map(allResults => {
+          (allResults.head, allResults(1)) match {
+            case (Right(nearlineResults), Right(onlineResults)) =>
+              if (nearlineResults.length < 10000 && onlineResults.length < 10000) {
+                logger.info(s"About to send bulk messages for ${nearlineResults.length} nearline results")
+                framework.bulkSendMessages(routingKey, nearlineResults)
+                logger.info(s"About to send bulk messages for ${onlineResults.length} online results")
+                framework.bulkSendMessages(routingKey, onlineResults)
+                logger.info(s"Bulk messages sent; about to send the RestorerSummaryMessage for project ${updateMessage.id}")
+                val msg = RestorerSummaryMessage(updateMessage.id, ZonedDateTime.now(), updateMessage.status, numberOfAssociatedFilesNearline = nearlineResults.length, numberOfAssociatedFilesOnline = onlineResults.length)
+                Right(MessageProcessorConverters.contentToMPRV(msg.asJson))
+              } else {
+                throw new RuntimeException(s"Too many files attached to project ${updateMessage.id}, nearlineResults = ${nearlineResults.length}, onlineResults = ${onlineResults.length}")
+              }
+            case (Left(nearlineErr), _) =>
+              logger.error(s"Could not connect to Matrix store: $nearlineErr")
```

Comment:
This is a bit tricky to read, I'd put the whole body of the map into its own function:

```scala
Future.sequence(
    Seq(getNearlineResults(updateMessage.id), getOnlineResults(updateMessage.id))
).map(processResults)
```

with

```scala
private def processResults(results:Seq[whatever]) = (allResults.head, allResults(1)) match {
.
.
.
etc.
.
.
.
}
```

Reply:
Alright, fair point